tests/disaster-recovery/sc_arbiter/test_zone_shutdown_and_crash.py

```diff
@@ -0,0 +1,514 @@
+import pytest
+import logging
+import time
+import random
+import concurrent.futures as futures
+from datetime import datetime, timezone
+
+from ocs_ci.helpers.stretchcluster_helper import (
+    recover_from_ceph_stuck,
+    recover_workload_pods_post_recovery,
+)
+from ocs_ci.ocs.resources.stretchcluster import StretchCluster
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.ocs.resources.pvc import get_pvc_objs
+from ocs_ci.ocs.node import wait_for_nodes_status, get_nodes
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.resources.pod import (
+    get_ceph_tools_pod,
+    wait_for_pods_to_be_in_statuses,
+    get_debug_pods,
+    get_not_running_pods,
+)
+from ocs_ci.helpers.sanity_helpers import Sanity
+from ocs_ci.ocs.exceptions import (
+    CommandFailed,
+    ResourceWrongStatusException,
+    CephHealthException,
+    UnexpectedBehaviour,
+)
+from ocs_ci.utility.retry import retry
+from ocs_ci.framework.pytest_customization.marks import stretch_cluster, turquoise_squad
+
+log = logging.getLogger(__name__)
+
+
+@stretch_cluster
+@turquoise_squad
+class TestZoneShutdowns:
+
+    zones = constants.ZONES_LABELS
+    zones.remove("arbiter")
+
+    @pytest.fixture()
+    def init_sanity(self, request, nodes):
+        """
+        Initial Cluster sanity
+        """
+        self.sanity_helpers = Sanity()
+
+        def finalizer():
+            """
+            Make sure all the nodes are Running and
+            the ceph health is OK at the end of the test
```

Comment:
you could replace all your teardown handling code with node_restart_teardown factory. please check

Reply:
I tested this but it doesnt work in case of parmetrized run.. it doesnt restart nodes until the whole test is completely executed for all the parameters. Hence sticking to this code

Reply:
I notice that the default scope of node_restart_teardown is set to 'function,' implying that the fixture will be executed once per test function and each combination of parameters generated by @pytest.mark.parametrize will result in a separate test invocation. IIUC, it should ideally work